src/components/lib/Makefile

```diff
@@ -125,3 +125,18 @@ distclean:
 		echo "*************************[TASK  STOP]*************************";\
 		echo ""; \
 	done
+
+LIB_DIRS := $(shell echo "$(REBUILD_DIRS)" | tr ' ' '\n' | grep '/lib/')
+component_rebuild:
+	$(info **************************************************)
+	$(info **********[ Rebuilding Related Libraries]*********)
+	$(info **************************************************)
+	$(info LIB_DIRS: $(LIB_DIRS))
+	@for dir in $(LIB_DIRS) ; do \
+		echo "*************************[TASK START]*************************";\
+		echo "                    [ Building $$dir ]";\
+		$(MAKE) PLATFORM=$(PLATFORM) $(MAKEOPTIONS) -C $$dir clean || exit $$?; \
```

Comment:
Ideally a `clean` wouldn't be required, as the dependencies should be properly updated, thus triggering a recompile only of the files that depend on the new header.

But I don't think that our makefiles are bug-free enough to rely on that ;-( Your invocation of `clean` here is likely required, even though it points to deep bugs in the build system.